packages/js/src/cache/preferences-cache.ts

```diff
@@ -3,11 +3,23 @@ import type { Cache } from './types';
 import { NovuEventEmitter, PreferenceEvents } from '../event-emitter';
 import { PreferenceLevel } from '../types';
 import { Preference } from '../preferences/preference';
+import { ListPreferencesArgs } from '../preferences/types';
 
 // these events should update the preferences in the cache
 const updateEvents: PreferenceEvents[] = ['preference.update.pending', 'preference.update.resolved'];
 
-const DEFAULT_KEY = 'default';
+const excludeEmpty = ({ tags }: ListPreferencesArgs) =>
+  Object.entries({ tags })
+    .filter(([_, value]) => value !== null && value !== undefined && !(Array.isArray(value) && value.length === 0))
```

Comment:
🤓 We can combine filter in the reduce function and save some performance from one less iteration.

Reply:
the object has only "one entry" - `tags`, so in this case, it will be always two iterations, one for the filter, and one for the reduce 🤓 but we can reduce it to one 😂